### PR TITLE
Field specials

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/field_specials.c [hunk 1]
```diff
@@ -0,0 +1,2503 @@
...
+    INCBIN_U16("data/field_specials/unk_83F5F90.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5FB0.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5FD0.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5FF0.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6010.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6030.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6050.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6070.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6090.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F60B0.gbap
```
Comment (jiangzhengwenjz) [inline src/field_specials.c]: right parenthesis on new line is not consistent with line 2253

Comment (jiangzhengwenjz) [inline src/field_specials.c]: Is it really better to explicitly write != FALSE here?

Comment (jiangzhengwenjz) [inline src/field_specials.c]: Right parenthesis on newline is not consistent with other instances

Comment (jiangzhengwenjz) [inline src/field_specials.c]: Is it really better to explicitly write != FALSE here?

Comment (PikalaxALT) [reply src/field_specials.c]: It appears to be the style of this particular programmer, based on his use of `!= TRUE` on the next line

Comment (PikalaxALT) [reply src/field_specials.c]: What other instances?

File: include/pokemon.h [hunk 2]
```diff
@@ -346,7 +346,7 @@ struct BattleTowerPokemon
              u32 gap:1;
              u32 abilityNum:1;
     /*0x1C*/ u32 personality;
-    /*0x20*/ u8 nickname[POKEMON_NAME_LENGTH + 1];
+    /*0x20*/ u8 nickname[11];
```
Comment (garakmon) [inline include/pokemon.h]: why did you change this

File: include/constants/flags.h [hunk 3]
```diff
@@ -1422,7 +1423,7 @@
 #define FLAG_WORLD_MAP_SIX_ISLAND_ALTERING_CAVE                     (SYS_FLAGS + 0xBE)
 #define FLAG_WORLD_MAP_SEVEN_ISLAND_TANOBY_RUINS_MONEAN_CHAMBER     (SYS_FLAGS + 0xBF)
 #define FLAG_WORLD_MAP_THREE_ISLAND_DUNSPARCE_TUNNEL                (SYS_FLAGS + 0xC0)
-#define FLAG_WORLD_MAP_SEVEN_ISLAND_SEAVAULT_CANYON_TANOBY_KEY      (SYS_FLAGS + 0xC1)
+#define FLAG_WORLD_MAP_SEVEN_ISLAND_SEVAULT_CANYON_TANOBY_KEY      (SYS_FLAGS + 0xC1)
```
Comment (garakmon) [inline include/constants/flags.h]: space

File: include/pokemon.h (hunk 2 above)
Comment (PikalaxALT) [reply include/pokemon.h]: Messed with the size of `struct BattleTowerEReaderTrainer`, defined in global.h
